docs/editing-components.md

```diff
@@ -0,0 +1,408 @@
+
+
+
+
+## The parts of a Wagtail block
+
+Blocks are implemented via several different bits of code:
+
+1. [Defining a block's fields and other properties in a Python class](#the-python-class)
+1. [Adding the class to a page's StreamField block options](#adding-it-to-a-streamfield)
+1. [Creating an HTML template for rendering the block on a page](#the-html-template)
+1. [(Optionally) adding some Less for styling the block](#adding-some-style)
+1. [(Optionally) adding some JavaScript for adding advanced behavior](#adding-some-javascript)
+
+Before you dive in further,
+[check out the Atomic Structure page](../atomic-structure/)
+and familiarize yourself with our basic concepts
+of atoms, molecules, and organisms.
+
+
+### The back end
+
+#### The Python class
+
+A component's fields and other properties are defined in a Python class,
+typically a subclass of Wagtail's
+[`StructBlock`](http://docs.wagtail.io/en/v1.13.4/topics/streamfield.html#structblock).
+These classes are located in a number of different files across the repository,
+but there are two major categories they fall into:
+
+1. Files corresponding to a general-purpose, site-wide atomic component.
+   These files—`atoms.py`, `molecules.py`, and `organisms.py`—are located in
+   [`cfgov/v1/atomic_elements`](https://github.com/cfpb/cfgov-refresh/tree/master/cfgov/v1/atomic_elements).
+2. Files that are specific to a particular sub-app, such as regulations3k's
+   [blocks.py](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/regulations3k/blocks.py).
+
+There are other places where StreamField block classes are defined
+(particularly blocks that are only ever used as fields within another block),
+but these are the two most common locations
+where top-level Wagtail modules are stored.
+
+A simple component class looks like this:
+
+```python
+class RelatedContent(blocks.StructBlock):                      # 1
+    heading = blocks.CharBlock(required=False)                 # 2
+    paragraph = blocks.RichTextBlock(required=False)           # 3
+    links = blocks.ListBlock(atoms.Hyperlink())                # 4
+
+    class Meta:                                                # 5
+        icon = 'grip'                                          # 6
+        label = 'Related content'                              # 7
+        template = '_includes/molecules/related-content.html'  # 8
+```
+
+There are a few things happening here:
+
+1. The `RelatedContent` class is a subclass of
+   [Wagtail's `StructBlock`](https://docs.wagtail.io/en/v1.13.4/topics/streamfield.html#structblock),
+   which allows for the combination of a fixed number of other sub-blocks
+   (see previous comment about blocks being nested within other blocks)
+   into a single unit (what we'd think of as a "module" in the Wagtail editor).
+   This one has three sub-blocks (lines 2, 3, and 4).
+2. The `heading` field uses the basic Wagtail `CharBlock`, which results in a
```

Comment:
these mentions of the Wagtail block types would be a great place to link to the Wagtail docs again:

https://docs.wagtail.io/en/v1.13.4/topics/streamfield.html#charblock
https://docs.wagtail.io/en/v1.13.4/topics/streamfield.html#richtextblock
https://docs.wagtail.io/en/v1.13.4/topics/streamfield.html#listblock

Reply:
Good idea! Addressed in 89f23454e89e7befda7ec3cbe69e2b4ca61cd59e